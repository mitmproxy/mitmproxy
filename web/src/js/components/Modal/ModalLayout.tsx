import * as React from "react"

type ModalLayoutProps = {
    children: React.ReactNode,
}

export default function ModalLayout ({ children}: ModalLayoutProps ) {
    return (
        <div>
        <div className="modal-backdrop fade in"></div>
        <div className="modal modal-visible" id="optionsModal" tabIndex={-1} role="dialog" aria-labelledby="options">
        <div className="modal-dialog modal-lg" role="document">
            <div className="modal-content">
                {children}
            </div>
        </div>
    </div>
        </div>
    )
}
