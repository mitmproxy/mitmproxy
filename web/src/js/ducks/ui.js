export const SELECT_FLOW = "SELECT_FLOW"
export const SET_ACTIVE_MENU = 'SET_ACTIVE_MENU'


const defaultState = {
    activeMenu: 'Start',
}
export default function reducer(state = defaultState, action) {
    switch (action.type) {
        case SET_ACTIVE_MENU:
            return {
                ...state,
                activeMenu: action.activeMenu
            }
        case SELECT_FLOW:
            let isNewSelect = (action.flowId && !action.currentSelection)
            let isDeselect = (!action.flowId && action.currentSelection)
            if(isNewSelect) {
                return {
                    ...state,
                    activeMenu: "Flow"
                }
            }
            if(isDeselect && state.activeMenu === "Flow") {
                return {
                    ...state,
                    activeMenu: "Start"
                }
            }
            return state
        default:
            return state
    }
}

export function setActiveMenu(activeMenu) {
    return {
        type: SET_ACTIVE_MENU,
        activeMenu
    }
}

