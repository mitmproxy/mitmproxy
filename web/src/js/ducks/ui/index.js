import { combineReducers } from 'redux'
import flow from './flow'
import header from './header'
import focus from './focus'
import prompt from './prompt'

export default combineReducers({
    flow,
    header,
    focus,
    prompt,
})
