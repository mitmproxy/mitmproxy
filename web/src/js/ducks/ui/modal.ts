export const HIDE_MODAL = "UI_HIDE_MODAL";
export const SET_ACTIVE_MODAL = "UI_SET_ACTIVE_MODAL";

const defaultState = {
    activeModal: undefined,
};

export default function reducer(state = defaultState, action) {
    switch (action.type) {
        case SET_ACTIVE_MODAL:
            return {
                ...state,
                activeModal: action.activeModal,
            };

        case HIDE_MODAL:
            return {
                ...state,
                activeModal: undefined,
            };
        default:
            return state;
    }
}

export function setActiveModal(activeModal) {
    return { type: SET_ACTIVE_MODAL, activeModal };
}

export function hideModal() {
    return { type: HIDE_MODAL };
}
