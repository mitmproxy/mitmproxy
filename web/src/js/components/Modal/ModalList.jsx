import React from 'react'
import ModalLayout from './ModalLayout'
import OptionContent from './OptionModal'
import CopyContent from './CopyModal'

function OptionModal() {
    return (
        <ModalLayout>
            <OptionContent/>
        </ModalLayout>
    )
}

function CopyModal() {
    return (
        <ModalLayout>
            <CopyContent/>
        </ModalLayout>
    )
}

export default [ OptionModal, CopyModal]
