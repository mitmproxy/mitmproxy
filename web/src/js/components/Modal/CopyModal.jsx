import React, { Component } from "react"
import { connect } from "react-redux"
import * as modalAction from "../../ducks/ui/modal"
import Clipboard from 'react-clipboard.js';
import * as flowsActions from "../../ducks/flows"


class PureCopyModal extends Component {

    constructor(props, context) {
        super(props, context)
        this.state = { title: 'Curl', curl: '' }
    }

    componentDidMount() {
        const { flow, copyFlow } = this.props
        copyFlow(flow)
        .then(res => res.json())
        .then((result) => {
            this.setState({
                curl: result.curl
            })
          }
        )
      }

    render() {
        const { hideModal} = this.props
        const { title, curl } = this.state
        return (
            <div>
                <div className="modal-header">
                    <button type="button" className="close" data-dismiss="modal" onClick={() => {
                        hideModal()
                    }}>
                        <i className="fa fa-fw fa-times"></i>
                    </button>
                    <div className="modal-title">
                        <h4>{ title }</h4>
                    </div>
                </div>

                <div className="modal-body">
                    <div style={{overflowWrap: 'break-word'}} id="curl">{curl}</div>
                </div>

                <div className="modal-footer">
                    <Clipboard className="btn btn-primary" data-clipboard-target="#curl">copy</Clipboard>
                </div>
            </div>
        )
    }
}

export default connect(
    state => ({
        flow: state.flows.byId[state.flows.selected[0]],
    }),
    {
        hideModal: modalAction.hideModal,
        copyFlow: flowsActions.copy
    }
)(PureCopyModal)
