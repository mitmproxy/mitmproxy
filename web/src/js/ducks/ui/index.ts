import { combineReducers } from "redux";
import flow from "./flow";
import modal from "./modal";
import optionsEditor from "./optionsEditor";

// TODO: Just move ducks/ui/* into ducks/?
export default combineReducers({
    flow,
    modal,
    optionsEditor,
});
