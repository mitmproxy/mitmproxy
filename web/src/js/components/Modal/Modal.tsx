import * as React from "react"
import ModalList from './ModalList'
import { useAppSelector } from "../../ducks";


export default function PureModal() {
    const activeModal : string = useAppSelector(state => state.ui.modal.activeModal)
    const ActiveModal:(() => JSX.Element)  | undefined= ModalList.find(m => m.name === activeModal )

    return(
        activeModal&&ActiveModal!==undefined ? <ActiveModal/> : <div/>
    )
}
