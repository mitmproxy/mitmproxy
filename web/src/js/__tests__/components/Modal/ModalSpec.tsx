import * as React from "react";
import Modal from "../../../components/Modal/Modal";
import { render } from "../../test-utils";
import { setActiveModal } from "../../../ducks/ui/modal";

test("Modal Component", async () => {
    const { asFragment, store } = render(<Modal />);
    expect(asFragment()).toMatchSnapshot();

    store.dispatch(setActiveModal("OptionModal"));
    expect(asFragment()).toMatchSnapshot();
});
