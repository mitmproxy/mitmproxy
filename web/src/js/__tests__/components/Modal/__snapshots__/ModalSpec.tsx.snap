// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;

exports[`Modal Component 2`] = `
<DocumentFragment>
  <div>
    <div
      class="modal-backdrop fade in"
    />
    <div
      aria-labelledby="options"
      class="modal modal-visible"
      id="optionsModal"
      role="dialog"
      tabindex="-1"
    >
      <div
        class="modal-dialog modal-lg"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div>
            <div
              class="modal-header"
            >
              <button
                class="close"
                data-dismiss="modal"
                type="button"
              >
                <i
                  class="fa fa-fw fa-times"
                />
              </button>
              <div
                class="modal-title"
              >
                <h4>
                  Options
                </h4>
              </div>
            </div>
            <div
              class="modal-body"
            >
              <div
                class="form-horizontal"
              >
                <div
                  class="form-group"
                >
                  <div
                    class="col-xs-6"
                  >
                    <label
                      for="anticache"
                    >
                      anticache
                    </label>
                    <div
                      class="help-block small"
                    >
                      Strip out request headers that might cause the server to return 304-not-modified.
                    </div>
                  </div>
                  <div
                    class="col-xs-6"
                  >
                    <div
                      class=""
                    >
                      <div
                        class="checkbox"
                      >
                        <label>
                          <input
                            name="anticache"
                            type="checkbox"
                          />
                          Enable
                        </label>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="form-group"
                >
                  <div
                    class="col-xs-6"
                  >
                    <label
                      for="body_size_limit"
                    >
                      body_size_limit
                    </label>
                    <div
                      class="help-block small"
                    >
                      Byte size limit of HTTP request and response bodies. Understands k/m/g suffixes, i.e. 3m for 3 megabytes.
                    </div>
                  </div>
                  <div
                    class="col-xs-6"
                  >
                    <div
                      class=""
                    >
                      <input
                        class="form-control"
                        name="body_size_limit"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="form-group"
                >
                  <div
                    class="col-xs-6"
                  >
                    <label
                      for="connection_strategy"
                    >
                      connection_strategy
                    </label>
                    <div
                      class="help-block small"
                    >
                      Determine when server connections should be established. When set to lazy, mitmproxy tries to defer establishing an upstream connection as long as possible. This makes it possible to use server replay while being offline. When set to eager, mitmproxy can detect protocols with server-side greetings, as well as accurately mirror TLS ALPN negotiation.
                    </div>
                  </div>
                  <div
                    class="col-xs-6"
                  >
                    <div
                      class=""
                    >
                      <select
                        class="form-control"
                        name="connection_strategy"
                      >
                        <option
                          value="eager"
                        >
                          eager
                        </option>
                        <option
                          value="lazy"
                        >
                          lazy
                        </option>
                      </select>
                    </div>
                  </div>
                </div>
                <div
                  class="form-group"
                >
                  <div
                    class="col-xs-6"
                  >
                    <label
                      for="listen_port"
                    >
                      listen_port
                    </label>
                    <div
                      class="help-block small"
                    >
                      Proxy service port.
                    </div>
                  </div>
                  <div
                    class="col-xs-6"
                  >
                    <div
                      class=""
                    >
                      <input
                        class="form-control"
                        name="listen_port"
                        type="number"
                        value="8080"
                      />
                    </div>
                    <div
                      class="small"
                    >
                      Default: 
                      <strong>
                         8080 
                      </strong>
                       
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="modal-footer"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
